### ISSUE TITLE
Unexpected item in KeyValue list when using Fable.Helpers.React.fs

### ISSUE THREAD

**Issue by Krzysztof-Cieslak** [fsprojects/Fable#129]
Sample code:
```fsharp
#r "../../node_modules/fable-core/Fable.Core.dll"
#load "../../node_modules/fable-import-react/Fable.Import.React.fs" 
#load "../../node_modules/fable-import-react/Fable.Helpers.React.fs"

open System
open Fable.Core
open Fable.Import
module R = Fable.Helpers.React
module P = Fable.Helpers.React.Props

let (^) = (<|)
 
ReactDom.render(
    R.span [ P.Value ^ U2.Case1 "Hello World" ] [],
    Browser.document.getElementById "content"
) 
```

Full error message:
```
Unexpected item in KeyValue list at (L14,11)-(L14,47): Apply
  (Apply
     (Apply
        (Apply
           (Value (TypeRef Code Module),[Value (StringConst "op_Concatenate")],
            ApplyGet,PrimitiveType (Function 0),Some {start = {line = 14;
                                                               column = 13;};
                                                      end = {line = 14;
                                                             column = 20;};}),[],
         ApplyMeth,PrimitiveType (Function 1),Some {start = {line = 14;
                                                             column = 13;};
                                                    end = {line = 14;
                                                           column = 20;};}),
      [Value
         (Lambda
            ([{name = "arg0";
               typ = DeclaredType U2 Union;}],
             Value
               (ArrayConst
                  (ArrayValues
                     [Value (StringConst "value");
                      Value (IdentValue {name = "arg0";
                                         typ = DeclaredType U2 Union;})],Tuple))))],
      ApplyMeth,DeclaredType U2 Union,Some {start = {line = 14;
                                                     column = 23;};
                                            end = {line = 14;
                                                   column = 45;};}),
   [Value (StringConst "Hello World")],ApplyMeth,DeclaredType HTMLAttr Union,
   Some {start = {line = 14;
                  column = 13;};
         end = {line = 14;
                column = 45;};}) 
```

Code without using `^` helper is working well (`R.span [ P.Value (U2.Case1 "Hello World") ] []`)

**Comment by ritcoder** [comment]
Not sure if it is the same issue but

This works

    let menuDisplayStyle (menu: Menu) state =
      if menu = state.activeMenu
      then "block"
      else "none"

	div [ 
	  ClassName "ui purple menu segment"
	  Width (U2.Case2 "300px")
	  Style [
		Display (menuDisplayStyle Menu.Calculator x.state)
		PaddingTop "10px"
		MarginLeft "-60px"
		MarginTop "18px"
		BorderBottomLeftRadius 0
		BorderBottomRightRadius 0
		BorderTopLeftRadius 0
		BorderTopRightRadius 0
	  ]
	] [
	  div [] [ unbox "something here"]
	]


This fails

    let menuDisplayStyle menu state =
      if menu = state.activeMenu
      then Display "block"
      else Display "none"

	div [ 
	  ClassName "ui purple menu segment"
	  Width (U2.Case2 "300px")
	  Style [
		menuDisplayStyle Menu.Calculator x.state
		PaddingTop "10px"
		MarginLeft "-60px"
		MarginTop "18px"
		BorderBottomLeftRadius 0
		BorderBottomRightRadius 0
		BorderTopLeftRadius 0
		BorderTopRightRadius 0
	  ]
	] [
	  div [] [ unbox "something here"]
	]

Error is

	Unexpected item in KeyValue list at (L78,20)-(L87,15): Sequential
	  ([VarDeclaration
		  ({name = "menu";
			typ = DeclaredType Menu Union;},
		   Apply
			 (Value (TypeRef Menu Union),[Value (StringConst "Calculator")],
			  ApplyCons,DeclaredType Menu Union,Some {start = {line = 79;
															   column = 33;};
													  end = {line = 79;
															 column = 48;};}),false);
		VarDeclaration
		  ({name = "state";
			typ = DeclaredType State Record;},
		   Apply
			 (Value This,[Value (StringConst "state")],ApplyGet,
			  DeclaredType State Record,Some {start = {line = 79;
													   column = 49;};
											  end = {line = 79;
													 column = 56;};}),false);
		Apply
		  (Apply
			 (Value This,[Value (StringConst "menuDisplayStyle")],ApplyGet,
			  PrimitiveType (Function 2),Some {start = {line = 79;
														column = 16;};
											   end = {line = 79;
													  column = 56;};}),
		   [Value (IdentValue {name = "menu";
							   typ = DeclaredType Menu Union;});
			Value (IdentValue {name = "state";
							   typ = DeclaredType State Record;})],ApplyMeth,
		   DeclaredType CSSProp Union,Some {start = {line = 79;
													 column = 16;};
											end = {line = 79;
												   column = 56;};})],
	   Some {start = {line = 79;
					  column = 16;};
			 end = {line = 79;
					column = 56;};}) (u:\Work\PlayGround\Fable\CreatingVSCodePlugin\test2\src\Components\System\TopPanel.fsx)

It appears a function that returns a `CSSProp` fails. Also `<|` fails.

**Comment by alfonsogarciacaro** [comment]
For performance the transformation was being done at compile time, so having the union case being determined at runtime didn't work. However it's true this is very limiting in many scenarios so I've made it that whenever a union case cannot be determined at compile time, the JS object will be built at runtime.

I hope this works for you, please tell me if you've any comment.

**Comment by ritcoder** [comment]
That seems fine. Will wait for the next update to check it out.

**Comment by alfonsogarciacaro** [comment]
I just published v0.3.1, you can try that one :+1:

**Comment by ritcoder** [comment]
It works. Thanks.
